src/Shared/TryParseMethodCache.cs

```diff
@@ -15,35 +15,63 @@
 
 namespace Microsoft.AspNetCore.Http
 {
-    internal static class TryParseMethodCache
+    internal class TryParseMethodCache
     {
-        private static readonly MethodInfo EnumTryParseMethod = GetEnumTryParseMethod();
+        private readonly MethodInfo _enumTryParseMethod;
 
         // Since this is shared source, the cache won't be shared between RequestDelegateFactory and the ApiDescriptionProvider sadly :(
-        private static readonly ConcurrentDictionary<Type, Func<Expression, MethodCallExpression>?> MethodCallCache = new();
-        internal static readonly ParameterExpression TempSourceStringExpr = Expression.Variable(typeof(string), "tempSourceString");
+        private readonly ConcurrentDictionary<Type, Func<Expression, Expression>?> _methodCallCache = new();
 
-        public static bool HasTryParseMethod(ParameterInfo parameter)
+        internal readonly ParameterExpression TempSourceStringExpr = Expression.Variable(typeof(string), "tempSourceString");
+
+        public TryParseMethodCache() : this(preferNonGenericEnumParseOverload: false)
+        {
+        }
+
+        public TryParseMethodCache(bool preferNonGenericEnumParseOverload)
```

Comment:
Is this only true in test code? If so, could you make a note of this? I imagine you wouldn't intentionally want this to be true.